tools/building/createTest.py

```diff
@@ -0,0 +1,147 @@
+
+
+def patharg(arg):
+    assert (arg)
+    if " " in arg:
+        raise argparse.ArgumentTypeError(
+            "The given suite name \"{}\" cannot contain spaces.".format(arg))
+    if "." in arg:
+        raise argparse.ArgumentTypeError(
+            "The given suite name \"{}\" cannot contain the file extensions.".
+            format(arg))
+    if not arg[0].isupper():
+        raise argparse.ArgumentTypeError(
+            "The given suite name \"{}\" must use CamelCase.".format(arg))
+    return arg
+
+
+def testname(arg):
+    assert (arg)
+    if " " in arg:
+        raise argparse.ArgumentTypeError(
+            "The given test name \"{}\" cannot contain spaces.".format(arg))
+    if "." in arg:
+        raise argparse.ArgumentTypeError(
+            "The given test name \"{}\" cannot contain the file extensions.".
+            format(arg))
+    if not arg[0].isupper():
+        raise argparse.ArgumentTypeError(
+            "The given test name \"{}\" must use CamelCase.".format(arg))
+    return arg
+
+
+PRECICE_TEST_BODY = """
+{
+  PRECICE_TEST(TODO);
+  precice::SolverInterface interface(context.name, context.config(), context.rank, context.size);
+
+  std::vector<VertexID> vertexIDs;
+
+  if (context.isNamed(TODO)) {
+    auto meshID = interface.getMeshID(TODO);
+    auto dataID = interface.getDataID(TODO, meshID);
+
+    std::vector<double> coords;
+
+    interface.setMeshVertices(meshID, TODO, coords.data(), vertexIDs.data());
+  } else {
+  }
+}
+"""
+
+
+def generateTestSource(name, suite, filepath):
+    if os.path.exists(filepath):
+        raise "The test source at \"{}\" already exists.".format(filepath)
+
+    includes = [
+        "<precice/SolverInterface.hpp>", "<vector>", '"testing/Testing.hpp"'
+    ]
+    suites = ["PreciceTests"] + suite
+    space = [""]
+    lines = ["#ifndef PRECICE_NO_MPI"]
+    lines += space
+    lines += ["#include " + inc for inc in includes if inc[0] != "<"]
+    lines += space
+    lines += ["#include " + inc for inc in includes if inc[0] == "<"]
+    lines += space
+    lines += ["BOOST_AUTO_TEST_SUITE({})".format(s) for s in suites]
+    lines += ["BOOST_AUTO_TEST_CASE({})".format(name), PRECICE_TEST_BODY]
+    lines += ["BOOST_AUTO_TEST_SUITE_END() // " + s for s in suites]
+    lines += space
+    lines += ["#endif // PRECICE_NO_MPI"]
+
+    with open(filepath, "w") as f:
+        f.writelines([line + "\n" for line in lines])
+
+
+def generateTestConfig(name, suite, filepath):
+    if os.path.exists(filepath):
+        print("The test config at \"{}\" already exists.".format(filepath))
+    else:
+        open(filepath, 'w').close()
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="preCICE integration test creation tool.")
+    parser.add_argument("suite",
+                        metavar="TestSuite",
+                        nargs="+",
+                        type=patharg,
+                        help="test suite(s) to add the test to")
```

Comment:
I would rather provide the suite as a real path (and not allow alternatives). This makes it easier to autocomplete with tab, if the script is executed from `tests`.

```
precice/tests$ python3 ../tools/building/createTest.py Serial/MultiCouplingFourSolvers Dummy
precice/tests$ python3 ../tools/building/createTest.py Serial/MultiCouplingFourSolvers/AnotherSuite Dummy
```
-> works. I autocompleted the path from the existing folder sturcture.

```
precice/tests$ python3 ../tools/building/createTest.py Serial MultiCouplingFourSolvers Dummy
```
-> also works, but I think we don't need it.

Reply:
Actually there is a bug with
```
precice/tests$ python3 ../tools/building/createTest.py Serial/MultiCouplingFourSolvers/AnotherSuite Dummy
```
I get the following in `Serial/MultiCouplingFourSolvers/AnotherSuite/Dummy.cpp`:
```
BOOST_AUTO_TEST_SUITE(Serial/MultiCouplingFourSolvers/AnotherSuite)
```

Reply:
If we use the relative path to denote the test suite, then we could also make it relative to the local directory.
This means that the suite could become an optional argument. 

```
$ cd tests/QN

$ createTest.py Subcyling QN1
Creates tests/QN/Subcyling/QN1.cpp

$ createTest.py Normal.cpp
Creates tests/QN/Normal.cpp

$ createTest.py Subcyling/Wavy QNWavy1
Creates tests/QN/Subcyling/Wavy/QNWavy1.cpp
```

Reply:
This issue is resolved at 4d040cf.